website/project/views/comment.py

```diff
@@ -15,104 +15,203 @@
 from website.notifications.emails import notify
 from website.filters import gravatar
 from website.models import Guid, Comment
+from website.addons.base import GuidFile
 from website.project.decorators import must_be_contributor_or_public
 from datetime import datetime
 from website.project.model import has_anonymous_link
+from website.project.views.node import _view_project, n_unread_comments
 
 
-def resolve_target(node, guid):
+@must_be_contributor_or_public
+def view_comments_project(auth, **kwargs):
+    """
+    Returns information needed to get comments for the total discussion page
+    """
+
+    node = kwargs['node'] or kwargs['project']
+    page = request.args.get('page', None)
+
+    OVERVIEW = "overview"
```

Comment:
@sloria @samanehsan  Here I've made an exception because <code>/\<pid\>/discussions?page=overview</code> returns all the comments on the overview page, but in this case, <code>comment.page == node</code>, not <code>overview</code> (I don't think <code>page=node</code> makes sense to users)